status/condition_set.go

```diff
@@ -163,16 +164,26 @@ func (c ConditionSet) SetTrueWithReason(conditionType string, reason, message st
 
 // SetUnknown sets the status of conditionType to Unknown and also sets the root condition
 // to Unknown if no other dependent condition is in an error state.
-func (r ConditionSet) SetUnknown(conditionType string) {
+func (r ConditionSet) SetUnknown(conditionType string) (modified bool) {
 	// set the specified condition
-	r.Set(Condition{
+	return r.Set(Condition{
 		Type:    conditionType,
 		Status:  metav1.ConditionUnknown,
 		Reason:  "AwaitingReconciliation",
 		Message: "object is awaiting reconciliation",
 	})
 }
 
+// SetFalse sets the status of t and the root condition to False.
+func (r ConditionSet) SetFalse(conditionType string, reason, message string) (modified bool) {
```

Comment:
nit: seems like we should match this function definition with the SetTrue() one since it takes in a reason
```suggestion
func (r ConditionSet) SetFalseWithReason(conditionType string, reason, message string) (modified bool) {
```

Reply:
Not willing to make this breaking change right now.